applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -688,51 +695,74 @@ static void audio_datapath_i2s_blk_complete(uint32_t frame_start_ts_us, uint32_t
 	}
 
 	/********** I2S RX **********/
-	static uint32_t *rx_buf;
 	static int prev_ret;
+	struct audio_data *rx_audio_block = NULL;
+	struct net_buf *buf_in;
 
 	if (IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL) || (CONFIG_AUDIO_DEV == GATEWAY)) {
-		/* Lock last filled buffer into message queue */
-		if (rx_buf_released != NULL) {
-			ret = data_fifo_block_lock(ctrl_blk.in.fifo, (void **)&rx_buf_released,
-						   BLOCK_SIZE_BYTES);
-
-			ERR_CHK_MSG(ret, "Unable to lock block RX");
+		if (rx_buf_released == NULL) {
+			/* No RX data available */
+			ERR_CHK_MSG(-ENOMEM, "No RX data available");
 		}
 
-		/* Get new empty buffer to send to I2S HW */
-		ret = data_fifo_pointer_first_vacant_get(ctrl_blk.in.fifo, (void **)&rx_buf,
-							 K_NO_WAIT);
-		if (ret == 0 && prev_ret == -ENOMEM) {
-			LOG_WRN("I2S RX continuing stream");
-			prev_ret = ret;
+		buf_in = net_buf_alloc(&pool_i2s_rx, K_NO_WAIT);
```

Comment:
After `buf_in` has been allocated, you'll need to free the buffer in any following error conditions

Reply:
All failing conditions leads to an ERR_CHK which will either halt or restart the system.